mlcolvar/core/transform/descriptors/eigs_adjacency_matrix.py

```diff
@@ -0,0 +1,119 @@
+import torch
+
+from mlcolvar.core.transform import Transform
+from mlcolvar.core.transform.descriptors.utils import compute_adjacency_matrix
+
+from typing import Union
+
+__all__ = ["EigsAdjMat"]
+
+class EigsAdjMat(Transform):
+    """
+    Eigenvalues of adjacency matrix transform, compute the eigenvalues of the adjacency matrix for a set of atoms from their positions
+    """
+
+    def __init__(self,
+                 mode : str,
+                 cutoff : float, 
+                 n_atoms : int,
+                 PBC: bool,
+                 cell: Union[float, list],
+                 scaled_coords : bool = False,
+                 switching_function = None) -> torch.Tensor:
+        """Initialize an eigenvalues of an adjacency matrix object.
+
+        Parameters
+        ----------
+        mode : str
+            Mode for cutoff application, either:
+            - 'continuous': applies a switching function to the distances which can be specified with switching_function keyword, has stable derivatives
+            - 'discontinuous': set at zero everything above the cutoff and one below, derivatives may be be incorrect
+        cutoff : float
+            Cutoff for the adjacency criterion 
+        n_atoms : int
+            Number of atoms in the system
+        PBC : bool
+            Switch for Periodic Boundary Conditions use
+        cell : Union[float, list]
+            Dimensions of the real cell, orthorombic-like cells only
+        scaled_coords : bool
+            Switch for coordinates scaled on cell's vectors use, by default False
+        switching_function : _type_, optional
+            Switching function to be applied for the cutoff, can be either initialized as a switching_functions/SwitchingFunctions class or a simple function, by default None
+
+        Returns
+        -------
+        torch.Tensor
+            Adjacency matrix of all the n_atoms according to cutoff
+        """
+        super().__init__(in_features=int(n_atoms*3), out_features=n_atoms)
+
+        # parse args
+        self.mode = mode
+        self.cutoff = cutoff 
+        self.n_atoms = n_atoms
+        self.PBC = PBC
+        self.cell = cell
+        self.scaled_coords = scaled_coords
+        self.switching_function = switching_function
+
+    def compute_adjacency_matrix(self, x):
+        x = compute_adjacency_matrix(pos=x,
+                                     mode = self.mode,
```

Comment:
I just know you'll hate me by the end (or beginning?) of the review, but PEP8 wants no space around ``=`` for function arguments.

Reply:
Don't worry I hate you already, I'll fix this nonetheless